#!/usr/bin/env python
PACKAGE = "vision"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# Params for bins
gen.add("binMinArea", int_t, 0, "Minimum area for a bin", 2000, 500, 50000)
gen.add("matchBound", double_t, 0, "match shape bound for bin", 0.1, 0.0, 1.0)
gen.add("ratioBound", double_t, 0, "edge ratio bound for bin", 2.2, 1.0, 3.0)
gen.add("blackThresh", int_t, 0, "For black bins thresholding", 150, 0, 255)
gen.add("adaptiveCoeff", double_t, 0, "For black bins thresholding", 2.0, 0, 10)
gen.add("adaptiveOffset", double_t, 0, "For black bins thresholding", 30, -30, 30)
gen.add("epsilon", int_t, 0, "approxPolyDP epsilon for bin", 20, 0, 50)

gen.add("loH1", int_t, 0, "Low H bound for aliens", 1, 0, 255)
gen.add("hiH1", int_t, 0, "High H bound for aliens", 25, 0, 255)
gen.add("loH2", int_t, 0, "Low H bound for aliens", 160, 0, 255)
gen.add("hiH2", int_t, 0, "High H bound for aliens", 180, 0, 255)

gen.add("loS1", int_t, 0, "Low S bound for aliens", 0, 0, 255)
gen.add("hiS1", int_t, 0, "High S bound for aliens", 255, 0, 255)

gen.add("loV1", int_t, 0, "Low V bound for aliens", 0, 0, 255)
gen.add("hiV1", int_t, 0, "High V bound for aliens", 255, 0, 255)

gen.add("alienMinArea", int_t, 0, "Minimum area for aliens", 700, 100, 50000)


exit(gen.generate(PACKAGE, "bins", "bins"))
